SharedPluginFiles/PluginUtils18_0.js

```diff
@@ -5,14 +5,34 @@ if (typeof FormIt === 'undefined')
 
 FormIt.PluginUtils = {};
 
-// Function to conver the given length into the current FormIt units.
+// convert the given length into the current FormIt units
 FormIt.PluginUtils.currentUnits = function(length)
 {
-    var kMetricFactor = 100.0 / (2.54 * 12.0);
-    if (FormIt.Model.GetUnitTypeCurrent() === FormIt.UnitType.kMetric)
-        return length * kMetricFactor;
+    var currentUnitType = FormIt.Model.GetUnitTypeCurrent();
 
-    return length;
+    // the default FormIt unit is 1 foot,
+    // so calculate all other units relative to that
+    var kImperialInchFactor = 1/12;
+    var kMetricMeterFactor = 100.0 / (2.54 * 12.0);
+    var kMetricCentimeterFactor = kMetricMeterFactor / 100;
+    var kMetricMillimeterFactor = kMetricMeterFactor / 1000;
+
+    switch (currentUnitType)
+    {
+        case FormIt.UnitType.kImperialInches:
+            return  length * kImperialInchFactor;
+        case FormIt.UnitType.kMetricMeter:
+            return  length * kMetricMeterFactor;
+        case FormIt.UnitType.kMetricCentimeter:
+            return  length * kMetricCentimeterFactor;
+        case FormIt.UnitType.kMetricMillimeter:
+            return  length * kMetricMillimeterFactor;
+        // if it's none of these, we're in feet, which is the default
+        // so return the length unchanged
+        default:
+            console.log("Did we get here?");
```

Comment:
Make a better error msg

Reply:
Whoops, this wasn't supposed to be there at all. Deleted.